Offroad-iOS/Offroad-iOS/Presentation/CourseQuest/ViewControllers/CourseQuestViewController.swift

```diff
@@ -0,0 +1,194 @@
+        setupControlsTarget()
+        
+        if let deadline = deadline {
+            let ddayString = Self.dday(from: deadline)
+            let dateString = formattedDate(from: deadline)
+            courseQuestView.deadlineDateLabel.text = "퀘스트 마감일: \(dateString)"
+            courseQuestView.ddayBadgeLabel.text = ddayString
+        }
+        
+        if let questId = questId {
+            fetchCourseQuestDetail(questId: questId)
+        }
+    }
+    
+    private func fetchCourseQuestDetail(questId: Int) {
+        courseQuestDetailService.getCourseQuestDetail(questId: questId) { [weak self] result in
+            switch result {
+            case .success(let dto):
+                DispatchQueue.main.async {
+                    self?.quests = dto.data.places
+                    self?.courseQuestView.courseQuestPlaceCollectionView.reloadData()
+                    //지도에 마커 표시
+                    self?.courseQuestView.configureMap(with: dto.data.places)
+                }
+            case .failure(let error):
+                print("❌ 코스 퀘스트 상세 정보 로드 실패:", error.localizedDescription)
+            }
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        guard let offroadTabBarController = self.tabBarController as? OffroadTabBarController else { return }
+        offroadTabBarController.hideTabBarAnimation()
+    }
+    
+    private func setupControlsTarget() {
+        courseQuestView.customBackButton.addTarget(self, action: #selector(customBackButtonTapped), for: .touchUpInside)
+    }
+    
+    /// 이 함수는 CourseQuestViewController 전용 D-Day 계산 함수입니다.
+    /// CourseQuestCollectionViewCell에도 유사한 함수가 존재하지만, QuestListViewController에서 D-Day를 표시하기 위한 용도이며,
+    /// 책임이 분리되어야 한다고 판단하여 별도로 정의했습니다.
+    private static func dday(from deadline: String?) -> String {
+        guard let deadline = deadline else {
+            return "D-?"
+        }
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+        formatter.locale = Locale(identifier: "ko_KR")
+        guard let deadlineDate = formatter.date(from: deadline) else {
+            return "D-?"
+        }
+        
+        let today = Calendar.current.startOfDay(for: Date())
+        let target = Calendar.current.startOfDay(for: deadlineDate)
+        let daysLeft = Calendar.current.dateComponents([.day], from: today, to: target).day ?? 0
+        
+        return daysLeft >= 0 ? "D-\(daysLeft)" : "종료"
+    }
+    
+    private func formattedDate(from deadline: String) -> String {
+        let inputFormatter = DateFormatter()
+        inputFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+        inputFormatter.timeZone = TimeZone(secondsFromGMT: 0)
+        
+        let outputFormatter = DateFormatter()
+        outputFormatter.dateFormat = "yy.MM.dd"
+        outputFormatter.locale = Locale(identifier: "ko_KR")
```

Comment:
P1
두 가지 문제가 있습니다.
- 서버에서 불러온 날짜(문자열)가 서울 시간대를 기준으로 한 것으로 알고 있는데, 
영국 기준으로 변환하신 것 같습니다!
두 formatter 모두 한국 시간으로 명시하여 통일하는 게 좋을 것 같습니다.

- 오브 앱에서 날짜 표현은 그레고리력 형식으로 통일됩니다.
formatter의 달력을 명시적으로 지정해주세요.

```suggestion
        let inputFormatter = DateFormatter()
        inputFormatter.calendar = .init(identifier: .gregorian)
        inputFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
        inputFormatter.timeZone = TimeZone(secondsFromGMT: 32400)
        
        let outputFormatter = DateFormatter()
        outputFormatter.calendar = .init(identifier: .gregorian)
        outputFormatter.dateFormat = "yy.MM.dd"
        outputFormatter.timeZone = TimeZone(secondsFromGMT: 32400)
        outputFormatter.locale = Locale(identifier: "ko_KR")
```

Reply:
피드백 감사합니다! 말씀해주신 두 가지 모두 확인했습니다.
서버에서 전달되는 날짜가 서울(UTC+9) 기준이라는 점을 고려하지 못했었네요..
제안해주신 코드 그대로 반영하겠습니다!